### PR TITLE
dictionary:   Improve StarDict access with stardict4j

### PR DESCRIPTION
**Pull request by miurahr**
- Use stardict4j access library
  - Stardict4j is created from OmegaT stardict code. 
  - Stardict4j supports .dict and dict.dz, .idx and idx.gz, as same as current OmegaT does
  - It can handle .syn synonym file.
  - It can handle 64bit index file as same as 32bit index which OmegaT can handle.
  - It can handle entry type such as 'm', 'x' and 'h'. 
  - It has true LRU cache for articles expiry by number of items and  time (depends Caffeine)

### CODE REVIEW

File: src/org/omegat/gui/preferences/view/DictionaryPreferencesPanel.form [hunk 1]
```diff
@@ -47,5 +47,16 @@
         <AuxValue name="JavaCodeGenerator_VariableModifier" type="java.lang.Integer" value="0"/>
       </AuxValues>
     </Component>
+    <Component class="javax.swing.JCheckBox" name="condensedViewCheckBox">
+      <Properties>
+        <Property name="text" type="java.lang.String" value="Use &amp;Condensed view"/>
+      </Properties>
```
Comment (miurahr) [inline src/org/omegat/gui/preferences/view/DictionaryPreferencesPanel.form:53]: The property should be localized such as;

```xml
        <Property name="text" type="java.lang.String" editor="org.netbeans.modules.i18n.form.FormI18nStringEditor">
          <ResourceString bundle="org/omegat/Bundle.properties" key="PREFS_DICTIONARY_CONDENSED" replaceFormat="OStrings.getString(&quot;{key}&quot;)"/>
        </Property>
```

File: src/org/omegat/gui/preferences/view/DictionaryPreferencesPanel.java [hunk 2]
```diff
@@ -60,10 +61,23 @@ private void initComponents() {
...
+                condensedViewCheckBoxActionPerformed(evt);
+            }
+        });
+        add(condensedViewCheckBox);
     }// </editor-fold>//GEN-END:initComponents
 
+    private void condensedViewCheckBoxActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_condensedViewCheckBoxActionPerformed
+        // TODO add your handling code here:
+    }//GEN-LAST:event_condensedViewCheckBoxActionPerformed
+
```
Comment (miurahr) [inline src/org/omegat/gui/preferences/view/DictionaryPreferencesPanel.java:77]: should not have action listener.

**Review by miurahr — COMMENTED** [review]
Need more change

**Comment by amake** [comment]
I am willing to take the stardict4j part as-is, but I would prefer the other parts to be a separate PR.

**Comment by miurahr** [comment]
> I am willing to take the stardict4j part as-is, but I would prefer the other parts to be a separate PR.

Done.

**Comment by amake** [comment]
Thanks. I've pushed to `master`.

**Pull request closed without merging**